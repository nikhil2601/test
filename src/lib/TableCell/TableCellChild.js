import styled from 'styled-components';

import { themeGet } from 'utils/theme';

const TableCellChild = styled.div`
    width: 100%;
    /**
     * Add all of the remaining styles from the theme
     */
    ${themeGet('TableCellChild.styles')};
`;

export default TableCellChild;
