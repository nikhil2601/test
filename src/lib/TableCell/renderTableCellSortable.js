import PropTypes from 'prop-types';
import React from 'react';
import _get from 'lodash/get';

import { ArrowIcon } from 'lib/SvgIcon';

function TableCellSortIcon(props) {
    const { sortSchema, col } = props;
    // this field comes from the schema
    const extraCss = _get(sortSchema, 'extraCss', '');

    return _get(sortSchema, 'orderBy') === (col.sortId || col.id) ? (
        <ArrowIcon width="18px" height="18px" extraCss={extraCss} />
    ) : null;
}

TableCellSortIcon.propTypes = {
    sortSchema: PropTypes.object,
    col: PropTypes.object,
};

TableCellSortIcon.defaultProps = {
    sortSchema: null,
    col: null,
};

function renderTableCellSortable(props, Component) {
    const { children, ...rest } = props;

    return (
        <Component {...rest}>
            {_get(props, 'sortSchema.iconPosition') === 'left' ? TableCellSortIcon(props) : null}
            {children}
            {_get(props, 'sortSchema.iconPosition') === 'left' ? null : TableCellSortIcon(props)}
        </Component>
    );
}

renderTableCellSortable.propTypes = {
    children: PropTypes.node,
};

renderTableCellSortable.defaultProps = {
    children: null,
};

export default renderTableCellSortable;
