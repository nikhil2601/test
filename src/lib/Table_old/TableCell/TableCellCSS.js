import _get from 'lodash/get';
import { css } from 'styled-components';

import SvgIcon from 'lib/SvgIcon';
import { getThemeProps } from 'utils/theme';
import { mergeObjects } from 'utils/merge';

const TableCellCSS = css`
    align-items: baseline;
    border-top: 1px solid ${getThemeProps('palette.common.lighter')};
    display: flex;
    padding: 16px 12px;
    position: relative;
    text-align: left;
    transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out;
    vertical-align: middle;
    /**
     * Display as a regular table cell above mobile view.
     */
    ${({ theme }) => theme.media.up('sm')`
        display: table-cell;
    `};
    /**
     * Add all of the remaining styles from the theme
     */
    ${getThemeProps('TableCell.styles')};
    /**
     * Style the inner SvgIcon component
     */
    ${SvgIcon} {
        transform: ${props => {
            const order = _get(props, 'order');
            const orderBy = _get(props, 'orderBy');
            const colId = _get(props, 'col.id');

            return orderBy === colId && order === 'desc' ? 'rotate(180deg)' : 'rotate(0deg)';
        }};
    }
    /**
     * Add in dynamic styles
     */
    ${({ cellProps, orientation, position, theme }) => {
        const cellType = cellProps.type || 'default';
        const orientationModePortrait = _get(orientation, 'mode') === 'portrait';

        const dynamicStyles = {};

        const portraitStyles = orientationModePortrait
            ? getThemeProps('TableCell.styles.portrait', null, {
                  theme,
              })
            : {};
        const positionStyles = getThemeProps(`TableCell.${position}.styles`, null, { theme });
        const cellTypeStyles = getThemeProps(`TableCell.${cellType}.styles`, {}, { theme });

        return mergeObjects(dynamicStyles, portraitStyles, cellTypeStyles, positionStyles);
    }};
`;

export default TableCellCSS;
