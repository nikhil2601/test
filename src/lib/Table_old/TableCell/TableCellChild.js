import styled from 'styled-components';

import { getThemeProps } from 'utils/theme';

const TableCellChild = styled.div`
    flex: 1 1 70%;
    display: inline-block;
    ${({ noDataAvailable }) => noDataAvailable && { width: '100%' }};
    /**
     * Add all of the remaining styles from the theme
     */
    ${getThemeProps('TableCellChild.styles')};
`;

export default TableCellChild;
