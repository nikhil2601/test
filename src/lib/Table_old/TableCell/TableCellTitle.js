import styled from 'styled-components';
import _get from 'lodash/get';

import { getThemeProps } from 'utils/theme';

const TableCellTitle = styled.div`
    color: #848484;
    display: block;
    flex: 1 1 30%;
    padding-right: 15px;
    text-align: right;
    /**
     * Don't display the title above mobile view.
     */
    ${({ theme }) => theme.media.up('sm')`
        display: none;
    `};
    /**
     * Add all of the remaining styles from the theme
     */
    ${getThemeProps('TableCellTitle.styles')};
    /**
     * If the orientation is in 'portrait' mode,
     * then force the display to be 'block' (visible)
     */
    ${({ orientation }) => ({
        display: _get(orientation, 'mode', 'landscape') === 'portrait' && 'block !important',
    })};
`;

export default TableCellTitle;
