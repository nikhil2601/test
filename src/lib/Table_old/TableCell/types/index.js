import TableCellButton from './TableCellButton';
import TableCellCheckbox from './TableCellCheckbox';
import TableCellHead from './TableCellHead';
import TableCellLink from './TableCellLink';
import TableCellStatus from './TableCellStatus';
import TableCellSvg from './TableCellSvg';
import TableCellDate from './TableCellDate';

const types = {
    button: TableCellButton,
    checkbox: TableCellCheckbox,
    date: TableCellDate,
    head: TableCellHead,
    link: TableCellLink,
    status: TableCellStatus,
    svg: TableCellSvg,
};

export default types;

export {
    TableCellButton,
    TableCellCheckbox,
    TableCellHead,
    TableCellLink,
    TableCellStatus,
    TableCellSvg,
};
