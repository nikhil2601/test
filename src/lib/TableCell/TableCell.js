import PropTypes from 'prop-types';
import React from 'react';
import _get from 'lodash/get';
import _isEmpty from 'lodash/isEmpty';
import _omit from 'lodash/omit';
import compose from 'recompose/compose';
import styled, { withTheme } from 'styled-components';

import Tooltip from 'lib/Tooltip';
import defaultCellTypes from 'lib/tableCellTypes';
import defaultTooltipTypes from 'lib/tooltipTypes';
import { callFunc, getHandler } from 'utils/actions';
import { checkObjectKeys } from 'utils/object';
import { responsiveProptypes } from 'utils/proptypes';
import { withActionHandlers } from 'lib/bindings';

import TableCellCSS from './TableCellCSS';
import TableCellChild from './TableCellChild';
import renderTableCellSortable from './renderTableCellSortable';

const TableCellStyled = styled.td`
    ${TableCellCSS};
`;

// eslint-disable-next-line valid-jsdoc
/**
 * A styled table-cell component.
 */
function TableCell(props) {
    const {
        cellProps,
        cellTypes,
        children,
        col,
        handlers,
        noDataAvailable,
        row,
        schema,
        setRef,
        tooltipTypes,
        type: typeProps,
    } = props;
    // Extract properties from the props.
    const helpText = _get(col, 'helpText');
    const dynamicHelpText = Boolean(typeof helpText === 'string');
    const helpTextProps = dynamicHelpText ? { tooltip: { id: helpText } } : helpText;
    // Combine the `cellTypes` from the context along with the default `cellTypes`.
    const tableCellTypes = {
        ...defaultCellTypes,
        ...cellTypes,
    };
    // Combine the `tooltipTypes` from the context along with the default `tooltipTypes`.
    const tooltipsList = {
        ...defaultTooltipTypes,
        ...tooltipTypes,
    };
    // Rebuild the `context` for the cell from the current props.
    const cellContext = {
        cellTypes,
        col,
        schema,
    };
    // Rebuild the `Col` properties for the cell.
    const colProps = {
        flex: _get(col, 'flex'),
        order: _get(col, 'order'),
        size: _get(col, 'size'),
    };
    // Extract the keysMap from the `col` for disabling the table cell.
    const disabledKeysMap = _get(col, 'disabled');
    // Should the table cell be disabled?
    const disabled = checkObjectKeys(row, disabledKeysMap, handlers);
    // Determine the cellType of table-cell to render.
    const cellType = _get(cellProps, 'cellType');
    // NOTE: Removing the big data props from the TableCellStyled.
    // It just pollutes the DOM.
    const componentProps = {
        disabled,
        ..._omit(props, [
            'baseUrl',
            'cellTypes',
            'data',
            'helpText',
            'noDataAvailable',
            'schema',
            'selectable',
            'setRef',
        ]),
        cellContext,
        cellType,
        ...colProps,
    };
    const TableCellRef = !tableCellTypes[cellType] ? setRef : null;
    const cellValueFormatterRef = _get(cellProps, 'formatter');
    // Get cell value if the formatter is passed in cellProps
    if (cellValueFormatterRef) {
        const cellValueFormatter = getHandler(handlers, cellValueFormatterRef);
        cellProps.value = callFunc(cellValueFormatter, { row });
    }

    // Build the children fro the TableCell.
    function renderChildren() {
        if (typeProps !== 'head' && cellType && tableCellTypes[cellType]) {
            const Cell = tableCellTypes[cellType];

            return <Cell {...componentProps} setRef={setRef} />;
        }
        return children;
    }
    // If the cellType is 'sortable' then we'll render the sortable `TableCell`.
    if (cellType === 'sortable') {
        return renderTableCellSortable(componentProps, TableCellStyled);
    }
    // We need to render the cell inside a row if we're
    // Build the final `TableCell` output.
    return (
        <TableCellStyled {...componentProps} ref={TableCellRef}>
            {!_isEmpty(helpText) ? (
                <Tooltip
                    content={renderChildren()}
                    {...helpTextProps}
                    tooltipProps={{ col, ...cellProps, ...componentProps }}
                    tooltipsList={tooltipsList}
                >
                    <TableCellChild noDataAvailable={noDataAvailable}>
                        {renderChildren()}
                    </TableCellChild>
                </Tooltip>
            ) : (
                <TableCellChild noDataAvailable={noDataAvailable}>
                    {renderChildren()}
                </TableCellChild>
            )}
        </TableCellStyled>
    );
}

TableCell.propTypes = {
    /**
     * Sets the `text-align` css property.
     */
    align: PropTypes.string,
    cellProps: PropTypes.object,
    cellTypes: PropTypes.object,
    children: PropTypes.node,
    /**
     * Properties of the current table-column
     */
    col: PropTypes.object,
    /**
     * Defines the 'flex' style property.
     */
    flex: responsiveProptypes(PropTypes.oneOfType([PropTypes.string])),
    /**
     * Should the `theme.grid.gutterWidth` be taken into account?
     */
    gutter: PropTypes.bool,
    /**
     * A map of action handlers for the `TableCell`.
     */
    handlers: PropTypes.object,
    noDataAvailable: PropTypes.bool,
    /**
     * Defines the 'flex-order' style property.
     */
    order: responsiveProptypes(PropTypes.number),
    /**
     * The row data for the `TableCell` when rendered inside a `DataList`.
     */
    row: PropTypes.object,
    schema: PropTypes.object,
    setRef: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
    /**
     * Specifies the size of the `column`.
     * Can be set to `auto` for an auto-sizing.
     * Can be set to `*` for sizing the `column` as normal. (default)
     * Can be set to any number from `1 - n` (n being the `theme.grid.column` size).
     * Can be set to an object for responsive styling. `{{ sm: 6, lg: 12 }}`
     */
    size: responsiveProptypes(
        PropTypes.oneOfType([
            PropTypes.oneOf(['auto', '*', false]),
            PropTypes.number,
            PropTypes.string,
        ])
    ),
    /**
     * Determines if the table-cell is sortable.
     */
    sortable: PropTypes.bool,
    tooltipTypes: PropTypes.object,
    /**
     * Determines the type of table cell to render.
     */
    type: PropTypes.oneOf(['head', 'body', 'foot']),
};

TableCell.defaultProps = {
    align: null,
    cellProps: null,
    cellTypes: null,
    children: null,
    col: {},
    flex: null,
    gutter: false,
    handlers: {},
    noDataAvailable: null,
    order: null,
    row: {},
    schema: {},
    setRef: null,
    size: null,
    sortable: false,
    tooltipTypes: null,
    type: null,
};

export default compose(withActionHandlers, withTheme)(TableCell);
