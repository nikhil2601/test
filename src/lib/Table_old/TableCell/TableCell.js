import PropTypes from 'prop-types';
import React from 'react';
import _get from 'lodash/get';
import _isEmpty from 'lodash/isEmpty';
import _omit from 'lodash/omit';
import compose from 'recompose/compose';
import styled from 'styled-components';

import Tooltip from 'lib/Tooltip';
import { ArrowIcon } from 'lib/SvgIcon';
import { disableTableCell } from 'utils/table';
import { withActionHandlers } from 'lib/bindings';

import TableCellCSS from './TableCellCSS';
import TableCellChild from './TableCellChild';
import TableCellTitle from './TableCellTitle';
import defaultCellTypes from './types';
import withTableContext from '../withTableContext';

/**
 * Simple table-cell
 *
 * @type {Function}
 */
const TableCellStyled = styled.td`
    ${TableCellCSS};
`;

/**
 * Render an arrow icon based on the current order and column being equal
 *
 * @method renderArrowIcon
 * @param  {string}        orderBy The ordeby value
 * @param  {Object}        col     The column info
 * @return {JSX|null}
 */
const renderArrowIcon = (
    { orderBy, col } // eslint-disable-line react/prop-types
) => (orderBy === col.id ? <ArrowIcon style={{ width: 18, height: 18 }} /> : null);

/**
 * Render a sortable cell with an arrow icon
 *
 * @method renderSortableCell
 * @param  {Object}            props     The props for the component
 * @param  {Function}          Component The component to render
 * @return {JSX}
 */
const renderSortableCell = (props, Component) => (
    <Component {...props}>
        {_get(props, 'sortSchema.iconPosition') === 'left' ? renderArrowIcon(props) : null}
        {_get(props, 'children')}
        {_get(props, 'sortSchema.iconPosition') === 'left' ? null : renderArrowIcon(props)}
    </Component>
);

const TableCell = props => {
    const {
        baseUrl,
        cellProps,
        cellTypes,
        children,
        col,
        handlers,
        helpText,
        noDataAvailable,
        oAuthToken,
        orientation,
        position,
        row,
        schema,
        setRef,
        tooltipsList,
    } = props;
    // Combine the `cellTypes` from the context along with the default `cellTypes`.
    const tableCellTypes = {
        ...defaultCellTypes,
        ...cellTypes,
    };
    // Rebuild the `context` for the cell from the current props.
    const cellContext = {
        baseUrl,
        cellTypes,
        handlers,
        helpText,
        oAuthToken,
        schema,
    };
    // Should the table cell be disabled?
    const disabled = disableTableCell(props);
    // NOTE: Removing the big data prop from the TableCellStyled.
    // It just pollutes the DOM.
    const componentProps = {
        disabled,
        ..._omit(props, [
            'actions',
            'baseUrl',
            'cellTypes',
            'data',
            'handlers',
            'helpText',
            'noDataAvailable',
            'oAuthToken',
            'schema',
            'selectable',
            'setRef',
        ]),
        cellContext,
    };
    // Determine the type of table-cell to render.
    const type = _get(cellProps, 'type');
    // Determine the component to use based on the position
    // of the table cell in the table.
    // Usually only used for table header since we use 'th' instead of 'td'
    const Component = position === 'head' ? tableCellTypes['head'] : TableCellStyled;

    // Build the children for the TableCell.
    const getChildren = () => {
        if (type && tableCellTypes[type]) {
            const Cell = tableCellTypes[type];
            return <Cell {...componentProps} setRef={setRef} />;
        } else {
            return children;
        }
    };

    // If the type is 'sortable' then we'll render the sortable TableCell.
    if (type === 'sortable') {
        return renderSortableCell(componentProps, Component);
    }

    // Return the final Table Cell Component.
    return (
        <Component {...componentProps} setRef={type === 'default' && setRef}>
            {_get(orientation, 'header') !== false && (
                <TableCellTitle orientation={orientation}>
                    {col.title}
                    {' :'}
                </TableCellTitle>
            )}
            {!_isEmpty(helpText) && typeof helpText === 'object' ? (
                <Tooltip
                    content={getChildren()}
                    {...helpText}
                    tooltipProps={{ baseUrl, col, oAuthToken, row }}
                    tooltipsList={tooltipsList}
                >
                    <TableCellChild orientation={orientation} noDataAvailable={noDataAvailable}>
                        {getChildren()}
                    </TableCellChild>
                </Tooltip>
            ) : (
                <TableCellChild orientation={orientation} noDataAvailable={noDataAvailable}>
                    {getChildren()}
                </TableCellChild>
            )}
        </Component>
    );
};

TableCell.propTypes = {
    /**
     * A base URL for making the API calls.
     */
    baseUrl: PropTypes.string,
    /**
     * What type of table cell are we rendering?
     * We can re-define the type of cell, it's value,
     * or the internal key it's associated with.
     * If we provide a key, then the table cell will
     * try to extract the key from its internal row-data.
     * Defaults to 'default' / 'text'.
     */
    cellProps: PropTypes.shape({
        key: PropTypes.string,
        type: PropTypes.string,
        value: PropTypes.string,
    }),
    cellTypes: PropTypes.object,
    /**
     * Table cell contents
     */
    children: PropTypes.node,
    /**
     * Properties of the current table-column
     */
    col: PropTypes.object,
    handlers: PropTypes.object,
    /**
     * Add a Tooltip component to the TableCell.
     */
    helpText: PropTypes.object,
    /**
     * A boolean representing that no data is available for the table.
     */
    noDataAvailable: PropTypes.bool,
    /**
     * An oAuth token to make API calls.
     */
    oAuthToken: PropTypes.string,
    /**
     * The orientation of the table, 'portrait' or 'landscape'
     */
    orientation: PropTypes.object,
    /**
     * Where are we positioning the table cell? head, body, or footer?
     * Defaults to 'body'.
     */
    position: PropTypes.oneOf(['head', 'body', 'footer']),
    /**
     * Properties of the current table-row
     */
    row: PropTypes.object,
    schema: PropTypes.object,
    /**
     * Set the ref attribute on the current table cell
     */
    setRef: PropTypes.object,
    /**
     * A list of tooltip components.
     */
    tooltipsList: PropTypes.array,
};

TableCell.defaultProps = {
    baseUrl: null,
    cellProps: {
        type: 'default',
    },
    cellTypes: null,
    children: null,
    col: {},
    handlers: null,
    helpText: null,
    noDataAvailable: false,
    oAuthToken: null,
    orientation: {},
    position: 'body',
    row: {},
    schema: null,
    setRef: null,
    tooltipsList: null,
};

export default compose(withTableContext, withActionHandlers)(TableCell);
