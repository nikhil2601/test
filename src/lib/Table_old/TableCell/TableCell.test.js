import React from 'react';
import { shallow } from 'enzyme';

import TableCell from './index';

describe('<TableCell />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<TableCell />);
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
