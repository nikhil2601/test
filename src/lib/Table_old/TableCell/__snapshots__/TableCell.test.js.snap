// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableCell /> should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
